script/e2e-test.sh

```diff
@@ -17,28 +17,80 @@
 set -e
 
 ROOT_DIR=`cd "$( dirname "${BASH_SOURCE[0]}" )/.." >/dev/null && pwd`
-IMG_MODIFIER=${1:?}
-DEV_TAG=${2:?}
+DEV_TAG=${1:?}
+IMG_MODIFIER=${2:-""}
+HELM_TLS_FLAGS="--tls --tls-ca-cert ca.cert.pem --tls-cert helm.cert.pem --tls-key helm.key.pem"
 
 source $ROOT_DIR/script/libtest.sh
 
+# Setup tiller with TLS support
+openssl genrsa -out ./ca.key.pem 4096
+cat <<EOF >> tls_config
+[ req ]
+distinguished_name="req_distinguished_name"
+prompt="no"
+
+[ req_distinguished_name ]
+C="ES"
+ST="Andalucia"
+L="Sevilla"
+O="Kubeapps"
+CN="localhost"
+EOF
+openssl req -key ca.key.pem -new -x509 -days 7300 -sha256 -out ca.cert.pem -config tls_config
+
+## server key
+openssl genrsa -out ./tiller.key.pem 4096
+## client key
+openssl genrsa -out ./helm.key.pem 4096
+
+openssl req -key tiller.key.pem -new -sha256 -out tiller.csr.pem -config tls_config
+openssl req -key helm.key.pem -new -sha256 -out helm.csr.pem -config tls_config
+
+openssl x509 -req -CA ca.cert.pem -CAkey ca.key.pem -CAcreateserial -in tiller.csr.pem -out tiller.cert.pem
+openssl x509 -req -CA ca.cert.pem -CAkey ca.key.pem -CAcreateserial -in helm.csr.pem -out helm.cert.pem
```

Comment:
what if we just generated a cert and put it in the repo or circleci env? Surely running all these commands increases the time for this test?